biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts

```diff
@@ -179,24 +207,33 @@ export class Biogrid implements Grid {
         } else {
           continue;
         }
-        energyUser.startCharging(energyUserReq);
+        (energyUser as BioBattery).startCharging(energyUserReq);
         clonedGraph.setNode(energyUser.gridItemName, energyUser);
       } else if (typeOldGridItem === GRID_ITEM_NAMES.LARGE_BATTERY) {
-        const energyUser = oldGridItem as BioBattery;
-        const energyUserReq =
-          energyUser.getMaxCapacity() - energyUser.getEnergyInJoules();
-        const typeSupplyingGridItem = this.getGridItemType(supplyingGridItem);
         if (typeSupplyingGridItem === GRID_ITEM_NAMES.SOLAR_PANEL) {
           const solarpanel = supplyingGridItem as SolarPanel;
           solarpanel.supplyPower(energyUserReq);
         } else {
           continue;
         }
-        energyUser.startCharging(energyUserReq);
+        (energyUser as BioBattery).startCharging(energyUserReq);
         clonedGraph.setNode(energyUser.gridItemName, energyUser);
       }
+      powerEdges.push({
+        v: supplyingGridItem.gridItemName,
+        w: energyUser.gridItemName,
+        // convert kilojoules into kilowatts
+        power: energyUserReq / (TIME.DISCRETE_UNIT_HOURS * 60 * 60),
```

Comment:
Energy from the battery is already in kilowatts, there is no need to change it.
If this is for solar panels, you can change it before `lines 215 - 216`. Or you can change it entirely in `Solar Panel class`

Reply:
Yeah this is fixed in #85, energy is in kilowatthour and is then converted to kilowatt by dividing by TIME.DISCRETE_UNIT_HOURS (which is currently 1)